.github/workflows/bump-go.yaml

```diff
@@ -0,0 +1,189 @@
+# support the target Go version.
+
+name: Bump Go
+
+on:
+  workflow_dispatch:
+    inputs:
+      target:
+        description: 'Go target version'
+        required: true
+        default: '1.15'
+      exclude:
+        description: 'Exclude regex'
+        required: false
+        default: "^third_party"
+      branch:
+        description: 'Branch name'
+        required: true
+        default: 'master'
+
+jobs:
+  meta:
+    runs-on: ubuntu-latest
+    outputs:
+      bump-include: ${{ steps.load-matrix.outputs.bump-include }}
+      bump-names: ${{ steps.load-matrix.outputs.bump-names }}
+    steps:
+    - name: Check out code onto GOPATH
+      uses: actions/checkout@v2
+    - name: Set up Go ${{ github.event.inputs.version }}.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ github.event.inputs.version }}.x
+
+    - id: load-matrix
+      run: |
+        go get github.com/bronze1man/yaml2json
+
+        function filtered_repos() {
+          local EXCLUDE="${1}"
+          local FILTER=$(cat "${EXCLUDE}" | yaml2json | jq "join(\"|\")")
+          if [ "$FILTER" = '""' ]; then
+            FILTER="\"match-nothing-${RANDOM}\""
+          fi
+          cat ${GITHUB_WORKSPACE}/repos.yaml | yaml2json | jq -c "map(select(.name | test(${FILTER}) | not))"
+        }
+
+        function filtered_names() {
+          filtered_repos "$1" | jq -c "map(.name)"
+        }
+
+        x=bump
+
+        echo "::group::Matrix names for $x"
+        filtered_names $x-exclude.yaml | jq .
+        echo "::endgroup::"
+
+        echo "::group::Matrix includes for $x"
+        filtered_repos $x-exclude.yaml | jq .
+        echo "::endgroup::"
+
+        echo "::set-output name=$x-include::$(filtered_repos $x-exclude.yaml)"
+        echo "::set-output name=$x-names::$(filtered_names $x-exclude.yaml)"
+
+  bump-go:
+    name: Bump Go 
+    needs: meta
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false # Keep running if one leg fails.
+      max-parallel: 5
+      matrix:
+        name: ${{fromJson(needs.meta.outputs.bump-names)}}
+        include: ${{fromJson(needs.meta.outputs.bump-include)}}
+
+    env:
+      GO111MODULE: on
+
+    steps:
+
+    - name: Set up Go ${{ github.event.inputs.version }}.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ github.event.inputs.version }}.x
+      id: go
+
+    - name: Check out code onto GOPATH
+      uses: actions/checkout@v2
+      with:
+        repository: ${{ matrix.name }}
+        ref: ${{ github.event.inputs.branch }}
+
+    - name: Install Dependencies
+      run: |
+        cd $(mktemp -d)
+        go get github.com/google/licenseclassifier
+        go get github.com/google/go-licenses
+        go get github.com/pierdipi/gobump/cmd/gobump 
+
+    - name: Bump Go
+      run: |
+        gobump --target ${{ github.event.inputs.version }} --exclude-regex ${{ github.event.inputs.exclude }}
```

Comment:
Can we just use `go mod edit -go=1.15`?

https://golang.org/ref/mod#go-mod-edit

Reply:
Yep, as long as we don't have any repository with nested modules, it should just work, AFAIK, we don't.

Reply:
I think that should be the goal state.

Reply:
Yes, nested modules are not encouraged from `golang` .

Reply:
Updated, PTAL.